pentaho/impl/src/main/java/pt/webdetails/cdf/dd/api/ResourcesApi.java

```diff
@@ -351,11 +360,13 @@ private Response.ResponseBuilder cacheControlHandler( IBasicFile file, Streaming
       responseBuilder.header( "Cache-Control", "max-age=0" );
       responseBuilder.header( "Etag", lastModifiedTime );
     } else {
-      IPluginResourceLoader resLoader = PentahoSystem.get( IPluginResourceLoader.class, null );
-
-      String maxAge = resLoader.getPluginSetting( this.getClass(), "max-age" );
-      if ( maxAge != null ) {
-        responseBuilder.header( "Cache-Control", "max-age=" + maxAge );
+      //Added to check PentahoSystem object for null values
+      if (PentahoSystem.get( IPluginResourceLoader.class, null ) != null) {
```

Comment:
This should mirror the code on lines 71-76

Reply:
@peterrinehart changes applied as noted